[bug] localStorage.getItem is not a function

### Is there an existing issue for this?

- [x] I have searched the existing issues

### RainbowKit Version

2.2.10

### wagmi Version

^2.19.3  or 2.19.5

### Current Behavior
Useing the `pnpm create @rainbow-me/rainbowkit@latest` from this repo Readme OR Using the script from the RainbowKit landing page `npm init @rainbow-me/rainbowkit@latest` gives me `next ^15.3.7, react ^19.1.3, react-dom: "^19.1.3"`
Using the Examples/With-Next-App under this repo gives me `next ^16.1.5, react ^19.2.3, react-dom ^19.2.3`
Both of the above give me 
```
    "@rainbow-me/rainbowkit": "2.2.10",
    "@tanstack/react-query": "^5.55.3",
    "viem": "2.38.0",
    "wagmi": "^2.19.3"
```
When using NodeJs v25.8.2 to run it in Google Chrome(with all extensions disabled), NextJs failed to run and I got those errors:
In the folder from the landing page script:
```
Module not found: Can't resolve "@react-native-async-storage/async-storage" 
in "/.../node_modules/@metamask/sdk/dist/browser/es"
[TypeError: localStorage.getItem is not a function]
```

In the Examples/With-Next-App folder under this repo:
``` 
TypeError: localStorage.getItem is not a function
    at ignore-listed frames {
  digest: '478258880'
}
unhandledRejection:  ReferenceError: indexedDB is not defined
```
My Google Chrome(with all extensions disabled): Version 146.0.7680.164 (Official Build) (64-bit)

NodeJs v24.14.0 can run NextJs but it still has `ReferenceError: indexedDB is not defined` error in the terminal

After adding `serverExternalPackages` in my next.config.js:
```
const nextConfig = {
	reactStrictMode: true,
	serverExternalPackages: [
		"pino",
		"@rainbow-me/rainbowkit",
		"viem",
		"wagmi",
		"pino-pretty",
	],
};
module.exports = nextConfig;
```
The new error appears:
```
⚠ ./src/app
Package @rainbow-me/rainbowkit can't be external
The request @rainbow-me/rainbowkit/styles.css matches serverExternalPackages (or the default list).
Only .mjs, .cjs, .js, .json, or .node can be handled by Node.js.


Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.
⨯ TypeError: Cannot read properties of null (reading 'useRef')
    at ignore-listed frames {
  digest: '4230347044'
}
[browser] Lit is in dev mode. Not recommended for production! See https://lit.dev/msg/dev-mode for more information.
[browser] Error checking Cross-Origin-Opener-Policy: HTTP error! status: 500 
```
Updating wagmi to 3.6.0 does not help.

### Expected Behavior

It should run without error

### Steps To Reproduce

Run the script from the RainbowKit landing page `npm init @rainbow-me/rainbowkit@latest` 
OR
Copy the code in the Examples/With-Next-App under this repo, and run `npm install`
Then run `npm run dev`

### Link to Minimal Reproducible Example (CodeSandbox, StackBlitz, etc.)

Examples/With-Next-App under this repo

### Anything else?

My Google Chrome(with all extensions disabled): Version 146.0.7680.164 (Official Build) (64-bit)

--- a/.changeset/ssr-safety-improvements.md
+++ b/.changeset/ssr-safety-improvements.md
@@ -2,4 +2,4 @@
 "@rainbow-me/rainbowkit": patch
 ---
 
-Improved SSR safety to prevent WalletConnect initialization warnings and avoid localStorage crashes in newer Node.js versions.
+Improved SSR safety to prevent WalletConnect initialization warnings and mitigate localStorage API availability changes in Node.js v25 and above.

--- a/.changeset/ssr-safety-improvements.md
+++ b/.changeset/ssr-safety-improvements.md
@@ -2,4 +2,4 @@
 "@rainbow-me/rainbowkit": patch
 ---
 
-Improved SSR safety to prevent WalletConnect initialization warnings.
+Improved SSR safety to prevent WalletConnect initialization warnings and avoid localStorage crashes in newer Node.js versions.

--- a/packages/rainbowkit/src/components/RainbowKitProvider/useFingerprint.ts
+++ b/packages/rainbowkit/src/components/RainbowKitProvider/useFingerprint.ts
@@ -3,7 +3,9 @@ import { useCallback, useEffect } from 'react';
 const storageKey = 'rk-version';
 
 function setRainbowKitVersion({ version }: { version: string }) {
-  localStorage.setItem(storageKey, version);
+  if (typeof window !== 'undefined') {
+    window.localStorage.setItem(storageKey, version);
+  }
 }
 
 export function useFingerprint() {

--- a/packages/rainbowkit/src/components/RainbowKitProvider/walletConnectDeepLink.ts
+++ b/packages/rainbowkit/src/components/RainbowKitProvider/walletConnectDeepLink.ts
@@ -7,7 +7,9 @@ export function setWalletConnectDeepLink({
   mobileUri: string;
   name: string;
 }) {
-  localStorage.setItem(
+  if (typeof window === 'undefined') return;
+
+  window.localStorage.setItem(
     storageKey,
     JSON.stringify({
       href: mobileUri.split('?')[0],
@@ -17,5 +19,7 @@ export function setWalletConnectDeepLink({
 }
 
 export function clearWalletConnectDeepLink() {
-  localStorage.removeItem(storageKey);
+  if (typeof window !== 'undefined') {
+    window.localStorage.removeItem(storageKey);
+  }
 }

--- a/packages/rainbowkit/src/transactions/transactionStore.ts
+++ b/packages/rainbowkit/src/transactions/transactionStore.ts
@@ -26,8 +26,8 @@ function safeParseJsonData(string: string | null): Data {
 
 function loadData(): Data {
   return safeParseJsonData(
-    typeof localStorage !== 'undefined'
-      ? localStorage.getItem(storageKey)
+    typeof window !== 'undefined'
+      ? window.localStorage.getItem(storageKey)
       : null,
   );
 }
@@ -203,7 +203,9 @@ export function createTransactionStore({
   }
 
   function persistData(): void {
-    localStorage.setItem(storageKey, JSON.stringify(data));
+    if (typeof window !== 'undefined') {
+      window.localStorage.setItem(storageKey, JSON.stringify(data));
+    }
   }
 
   function notifyListeners(): void {

--- a/packages/rainbowkit/src/utils/ens.ts
+++ b/packages/rainbowkit/src/utils/ens.ts
@@ -19,13 +19,13 @@ function safeParseJsonData(string: string | null): EnsData | null {
 }
 
 export function addEnsName(address: Address, ensName: string) {
-  if (!isAddress(address)) return;
+  if (!isAddress(address) || typeof window === 'undefined') return;
 
   const now = new Date();
 
   const expiry = new Date(now.getTime() + 180 * 60_000); // Set expiry to 3 hours from now
 
-  localStorage.setItem(
+  window.localStorage.setItem(
     getStorageEnsNameKey(address),
     JSON.stringify({
       ensName,
@@ -35,8 +35,10 @@ export function addEnsName(address: Address, ensName: string) {
 }
 
 export function getEnsName(address: Address): string | null {
+  if (typeof window === 'undefined') return null;
+
   const data = safeParseJsonData(
-    localStorage.getItem(getStorageEnsNameKey(address)),
+    window.localStorage.getItem(getStorageEnsNameKey(address)),
   );
 
   if (!data) return null;
@@ -44,14 +46,14 @@ export function getEnsName(address: Address): string | null {
   const { ensName, expires } = data;
 
   if (typeof ensName !== 'string' || Number.isNaN(Number(expires))) {
-    localStorage.removeItem(getStorageEnsNameKey(address));
+    window.localStorage.removeItem(getStorageEnsNameKey(address));
     return null;
   }
 
   const now = new Date();
 
   if (now.getTime() > Number(expires)) {
-    localStorage.removeItem(getStorageEnsNameKey(address));
+    window.localStorage.removeItem(getStorageEnsNameKey(address));
     return null;
   }
 

--- a/packages/rainbowkit/src/wallets/latestWalletId.ts
+++ b/packages/rainbowkit/src/wallets/latestWalletId.ts
@@ -1,15 +1,19 @@
 const storageKey = 'rk-latest-id';
 
 export function getLatestWalletId(): string {
-  return typeof localStorage !== 'undefined'
-    ? localStorage.getItem(storageKey) || ''
+  return typeof window !== 'undefined'
+    ? window.localStorage.getItem(storageKey) || ''
     : '';
 }
 
 export function addLatestWalletId(walletId: string): void {
-  localStorage.setItem(storageKey, walletId);
+  if (typeof window !== 'undefined') {
+    window.localStorage.setItem(storageKey, walletId);
+  }
 }
 
 export function clearLatestWalletId(): void {
-  localStorage.removeItem(storageKey);
+  if (typeof window !== 'undefined') {
+    window.localStorage.removeItem(storageKey);
+  }
 }

--- a/packages/rainbowkit/src/wallets/recentWalletIds.ts
+++ b/packages/rainbowkit/src/wallets/recentWalletIds.ts
@@ -10,8 +10,8 @@ function safeParseJsonArray<T>(string: string | null): T[] {
 }
 
 export function getRecentWalletIds(): string[] {
-  return typeof localStorage !== 'undefined'
-    ? safeParseJsonArray(localStorage.getItem(storageKey))
+  return typeof window !== 'undefined'
+    ? safeParseJsonArray(window.localStorage.getItem(storageKey))
     : [];
 }
 
@@ -22,5 +22,7 @@ function dedupe<T>(array: T[]): T[] {
 export function addRecentWalletId(walletId: string): void {
   const newValue = dedupe([walletId, ...getRecentWalletIds()]);
 
-  localStorage.setItem(storageKey, JSON.stringify(newValue));
+  if (typeof window !== 'undefined') {
+    window.localStorage.setItem(storageKey, JSON.stringify(newValue));
+  }
 }

